packages/client/lib/sync/txpool.ts

```diff
@@ -165,16 +177,38 @@ export class TxPool {
    * Adds a tx to the pool.
    *
    * If there is a tx in the pool with the same address and
-   * nonce it will be replaced by the new tx.
+   * nonce it will be replaced by the new tx, if it has a sufficent gas bump.
+   * (TODO) verifies certain constraints, if these are not matches, throws.
    * @param tx Transaction
    */
-  add(tx: TypedTransaction) {
+  async add(tx: TypedTransaction) {
```

Comment:
@ryanio do you think this method is a good central place for most checks? We can handle the throws internally here if we have a list of txs (remove them from list since these are invalid), but for RPC we catch those and instead throw the RPC error

Reply:
yes, but i'm thinking make a `validate(tx: TypedTransaction)` method in this class, and call into it at the top of add.

Reply:
Yep, makes sense, will do :+1: :smile: